rdtools/test/test_normalize_with_expected_power.py

```diff
@@ -0,0 +1,109 @@
+from pandas import Timestamp
+import numpy as np
+
+@pytest.fixture()
+def times_15():
+	return pd.date_range(start='20200101 12:00', end= '20200101 13:00', freq='15T')
+
+@pytest.fixture()
+def times_30():
+	return pd.date_range(start='20200101 12:00', end= '20200101 13:00', freq='30T')
+
+@pytest.fixture()
+def pv_15(times_15):
+	return pd.Series([1.0, 2.5, 3.0, 2.2, 2.1], index=times_15)
+
+@pytest.fixture()
+def expected_15(times_15):
+	return pd.Series([1.2, 2.3, 2.8, 2.1, 2.0], index=times_15)
+
+@pytest.fixture()
+def irradiance_15(times_15):
+	return pd.Series([1000.0, 850.0, 950.0, 975.0, 890.0], index=times_15)
+
+@pytest.fixture()
+def pv_30(times_30):
+	return pd.Series([1.0, 3.0, 2.1], index=times_30)
+
+@pytest.fixture()
+def expected_30(times_30):
+	return pd.Series([1.2, 2.8, 2.0], index=times_30)
+
+@pytest.fixture()
+def irradiance_30(times_30):
+	return pd.Series([1000.0, 950.0, 890.0], index=times_30)
+
+def test_normalize_with_expected_power_uniform_frequency(pv_15, expected_15, irradiance_15):
+	norm, insol = normalize_with_expected_power(pv_15, expected_15, irradiance_15)
+	expected_norm = pd.Series({Timestamp('2020-01-01 12:15:00', freq='15T'): 1.0,
+							   Timestamp('2020-01-01 12:30:00', freq='15T'): 1.0784313725490198,
+                               Timestamp('2020-01-01 12:45:00', freq='15T'): 1.0612244897959184,
+                               Timestamp('2020-01-01 13:00:00', freq='15T'): 1.0487804878048783})
+	expected_norm.name = 'energy_Wh'
+
+	expected_insol = pd.Series({Timestamp('2020-01-01 12:15:00', freq='15T'): 231.25,
+                                Timestamp('2020-01-01 12:30:00', freq='15T'): 225.0,
+                                Timestamp('2020-01-01 12:45:00', freq='15T'): 240.625,
+                                Timestamp('2020-01-01 13:00:00', freq='15T'): 233.125})
+	expected_insol.name = 'energy_Wh'
+
+	pd.testing.assert_series_equal(norm, expected_norm)
+	pd.testing.assert_series_equal(insol, expected_insol)
+
+def test_normalize_with_expected_power_energy_option(pv_15, expected_15, irradiance_15):
+	norm, insol = normalize_with_expected_power(pv_15, expected_15, irradiance_15, pv_input='energy')
+	expected_norm = pd.Series({Timestamp('2020-01-01 12:00:00', freq='15T'): np.nan,
+                               Timestamp('2020-01-01 12:15:00', freq='15T'): 5.714285714285714,
+                               Timestamp('2020-01-01 12:30:00', freq='15T'): 4.705882352941177,
+                               Timestamp('2020-01-01 12:45:00', freq='15T'): 3.5918367346938775,
+                               Timestamp('2020-01-01 13:00:00', freq='15T'): 4.097560975609756})
+	expected_norm.name = 'energy_Wh'
+
+	expected_insol = pd.Series({Timestamp('2020-01-01 12:15:00', freq='15T'): 231.25,
+ 								Timestamp('2020-01-01 12:30:00', freq='15T'): 225.0,
+ 								Timestamp('2020-01-01 12:45:00', freq='15T'): 240.625,
+ 								Timestamp('2020-01-01 13:00:00', freq='15T'): 233.125})
```

Comment:
From a user's perspective, it is a little strange that the returned normalized energy and insolation series have different lengths when `pv_input='energy'`.  Should we remove the leading nan on the normalized energy, or add one on the insolation?

Also, indentation for `expected_insol`

Reply:
That's a good point. I can imaging a bunch of edge cases here resulting in different behavior, I think the most conservative thing to do is to reindex both series to the union of their indexes.